public/api-docs/docs/user.yaml

```diff
@@ -0,0 +1,163 @@
+            $ref: '#/components/schemas/Token'
+      Token:
+        type: string
+        description: Authentication token
+        example: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NCwiZW1haWwiOiJ0aG9ueUBnbWFpbC5jb20iLCJpc0FkbWluIjpmYWxzZSwiaWF0IjoxNTYxMTYyMTY0LCJleHAiOjE1NjEyNDg1NjR9.jx2qbXyui03Dhb9F8iwwbMel9LMYIShLYr5Y5y94BtU
+      AuthToken:
+        type: string
+        description: bearer token sent along with request
+        example: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NCwiZW1haWwiOiJ0aG9ueUBnbWFpbC5jb20iLCJpc0FkbWluIjpmYWxzZSwiaWF0IjoxNTYxMTYyMTY0LCJleHAiOjE1NjEyNDg1NjR9.jx2qbXyui03Dhb9F8iwwbMel9LMYIShLYr5Y5y94BtU
+
+paths: 
+  /auth/signin: # signin endpoint
+    post: 
+      tags:
+        - User
+      description: 'Logs user into the system'
+      operationId: 'loginUser'
+      parameters: []
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#components/schemas/SigninUser'
+        required: true
+      responses: 
+        '200': 
+          description: Successful operation user login
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: integer
+                    description: 200 success status code
+                    example: '200'
+                  data:
+                    $ref: '#/components/schemas/SignedUser'
+        '400':
+          description: Response for an unsuccessful login
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: integer
+                    description: 400 error status code
+                    example: '400'
+                  error: 
+                    type: object/string
+                    example: 'Invalid email/password'
+  /auth/user: # authorize user endpoint
+    post:
+      tags:
+        - User
+      description: 'admin adds a new user to the app'
+      operationId: 'addUser'
+      parameters:
+        - name: Authorization
+          in: header
+          description: Authorization header with jwt token
+          required: true
+          schema:
+            $ref: '#/components/schemas/AuthToken'
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#components/schemas/AddUser'
+        required: true
+      responses:
+        '201':
+          description: user successfully created
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: integer
+                    description: 201 success status code
+                    example: '201'
+                  data:
+                    $ref: '#/components/schemas/User'
+        '409':
```

Comment:
please also add for input validations 422 and server error 500

Reply:
> I guess we now using yaml?

Yes, there was no way to split the `json` file into several files. Such action is not possible with json. It could only be done `yaml`.

Reply:
> please also add for input validations 422 and server error 500

ok